api/mutations/user/editUser.js

```diff
@@ -52,7 +51,35 @@ export default requireAuth(
       }
     }
 
-    const editedUser = await editUser(args, user.id);
+    if (input.email && typeof input.email === 'string') {
+      const pendingEmail = input.email;
+
+      // if user is changing their email, make sure it's not taken by someone else
+      if (pendingEmail !== currentUser.email) {
+        if (!isEmail(input.email)) {
+          return new UserError('Please enter a working email address');
+        }
+
+        const dbUsers = await getUsersByEmail(pendingEmail);
+        if (dbUsers && dbUsers.length > 0) {
+          return new UserError(
+            'That email address is already taken by another person on Spectrum.'
```

Comment:
Do we want to expose that information?

Reply:
We'll have to just keep an eye out if people bump into this without being malicious - in the meantime I'll obfuscate the error message so that it can't be determined.